src/js/dp/map.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * Requirements:
+ * `npm i -S mapbox-gl-js`
+ * Example setup:
+ * The map component requires 3 data attribute values:
+ * @param {String} style Either url or path name
+ * @param {Array<String>} center geolocation coordinates as a starting location for the map display 
+ * @param {Number} zoom Optional - default 6
+ * @param {String} token Optional - default undefined
+ * ```     
+ *    <div 
+ *       id="map"
+ *       data-style="mapbox://styles/mapbox/cjaudgl840gn32rnrepcb9b9g?access_code=<key>"
+ *       data-center="[51.42133794405771, -0.14646169096877365]"
+ *       data-zoom="6"
+ *       data-token="<PUBLIC_TOKEN>"
+ *   ></div> 
+ * ```
+ */
+let mapID = document.getElementById("map");
+let mapboxgl = null; // mapbox-gl-js 
+/** @internal */
+const _getMapStyle = (mapEl) =>  {
+    const mapStyle = mapEl.dataset.style;
+    if (!mapStyle) {
+        throw new Error("You must define a style value url or path!");
+    }
+    return mapStyle
+};
+/** @internal */
+const _getMapCenter = (mapEl) => {
+    let mapCenter = mapEl.dataset.center;
+    if (!mapCenter) {
+        throw new Error("You must declare a 'data-center' attribute");
+    }
+    mapCenter = mapCenter.replace(/\[|]/g, "");
+    if (mapCenter.length < 2) throw new Error(centerErrorMsg);
+    mapCenter = mapCenter.split(",");
+    mapCenter = mapCenter.map(center => {
```

Comment:
Counterintuitively the radix does not default to 10:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description

```suggestion
    mapCenter = mapCenter.map(center => parseInt(center, 10));
```